feat(KomikIndo): add komik indo presence

1intan99 opened the pull request:
## Description 
Adding Komik Indo Presence

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<h2> Website </h2>
<img src="https://i.imgur.com/dPCUgNK.png">
<h2> Discord Presence </h2>
<img src="https://i.imgur.com/vlrqb47.png">
<h2> Extension Settings </h2>
<img src="https://i.imgur.com/99aS1WB.png">
</details>

theusaf commented on websites/K/Komik Indo/metadata.json:
```diff
@@ -0,0 +1,30 @@
+	},
+	"url": "komikindo.tv",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/HccQ60t.png",
```
The [logo](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo) should be a 512x512px image.

theusaf commented on websites/K/Komik Indo/presence.ts:
```diff
@@ -0,0 +1,129 @@
+
+const enum Assets {
+	Logo = "https://i.imgur.com/HccQ60t.png",
+	Home = "https://i.imgur.com/m6Cw03l.png",
```
`Home` asset isn't used in the script, so it can be removed from the `Assets` enum

theusaf commented on websites/K/Komik Indo/presence.ts:
```diff
@@ -0,0 +1,129 @@
+	const presenceData: PresenceData = {
+			details: "Browsing...",
+			largeImageKey: Assets.Logo,
+			smallImageKey: Assets.Logo,
```
The `smallImageKey` might not be needed in this case, since it is just the logo.

theusaf commented on websites/K/Komik Indo/presence.ts:
```diff
@@ -0,0 +1,129 @@
+		{ pathname, href } = document.location,
+		buttons = await presence.getSetting<boolean>("buttons");
+	switch (true) {
+		case href === "https://komikindo.tv/":
```
```suggestion
		case pathname === "/":
```

theusaf replied:
Good point

theusaf commented on websites/K/Komik Indo/presence.ts:
```diff
@@ -0,0 +1,127 @@
+		{ pathname, href, hostname } = document.location,
+		buttons = await presence.getSetting<boolean>("buttons");
+	switch (true) {
+		case hostname === "komikindo.tv":
```
This condition is always true regardless of path. Perhaps revert it to the `href` check or similar.

theusaf commented on websites/K/Komik Indo/presence.ts:
```diff
@@ -0,0 +1,127 @@
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/HccQ60t.png",
```
```suggestion
	Logo = "https://i.imgur.com/YY8ZLqd.png",
```

1intan99 commented:
I've added the `getTitleText` function in this pull request to streamline the process of retrieving and cleaning text content from DOM elements. The primary reason for implementing this function is to improve code readability, maintainability, and efficiency. Here’s a detailed explanation:

- Code Reusability:
    - Instead of repeatedly writing the same code to query a DOM element and clean up its text content, I've encapsulated this logic within the `getTitleText` function. This reduces code duplication and makes it easier to manage updates in the future.
- Text Cleaning:
    - The function handles the cleaning of text by removing unnecessary characters like tabs `(\t)` and newlines `(\n)`. It also trims any leading or trailing whitespace. This ensures that the text used in the presence data is consistent and free of formatting artifacts that could affect the user experience.
- Fallback Handling:
   - In cases where the DOM element isn't found, the function returns `"Unknown"` instead of leaving the text undefined or null. This fallback mechanism helps maintain the stability of the presence data, preventing potential errors or issues in the application.
- Improved Readability:
   - By using `getTitleText`, the main presence update function is more concise and easier to read. The intent of the code is clearer when selecting and processing text content, making the overall logic more understandable to other developers.
   
Also, I've refactored the code from using a `switch(true)` statement to a more straightforward `if-else` structure. I compared `switch(true)` statement and `if-else` structure, I see that it is faster to detect using `if-else` than using `swtich(true)` statement.

Bas950 commented on websites/K/Komik Indo/metadata.json:
```diff
@@ -0,0 +1,30 @@
+		"en": "KomikIndo is an online comic reading site with the most complete and updated collection. You can read thousands of comic collections that we update every day for free. With a responsive and modern design, this website is the best place for those of you who want to read the latest Manga, Manhwa (Korean comics) and Manhua (Chinese comics)."
+	},
+	"url": "komikindo.tv",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/YY8ZLqd.png",
```
Bit of a blurry image?

1intan99 replied:
yea it a bit blurry, because i took it from the website icon and resize it to 512x512
